nix/overlay.nix

```diff
@@ -5,6 +5,17 @@ in {
     ocamlVersion.overrideScope' (oself: osuper: {
       lwt_react = osuper.lwt_react.overrideAttrs
         (o: { nativeBuildInputs = o.nativeBuildInputs ++ [ oself.cppo ]; });
+      lwt_domain = osuper.buildDunePackage {
+        pname = "lwt_domain";
+        src = prev.fetchFromGitHub {
+          owner = "ocsigen";
+          repo = "lwt";
+          rev = "654952b62dccea63f186747be79916dbfdccf48d";
+          sha256 = "sha256-j+ufbnovlcJnKM0DwU3Z/DRLeyJ9+AmhKSrJCpMVW7w=";
+        };
+        version = "dev";
+        propagatedBuildInputs = with osuper; [ domainslib lwt ];
+      };
```

Comment:
```suggestion
      lwt_domain = osuper.buildDunePackage {
        pname = "lwt_domain";
        version = "0.2.0-dev";
        inherit (osuper.lwt) src;
        propagatedBuildInputs = with osuper; [ domainslib lwt ];
      };
```

Reply:
This seems to work on my end and is closer to what we would upstream.

Reply:
Opened this PR to the overlays: https://github.com/anmonteiro/nix-overlays/pull/291

Reply:
This is merged in the overlays, if you need the very specific version you added then you can just override the `src` now, otherwise it should get the same version as the lwt version we have